command/ui.go

```diff
@@ -184,7 +188,9 @@ func (c *UiCommand) Run(args []string) int {
 	} else {
 		c.Ui.Output(fmt.Sprintf("Opening URL %q", url.String()))
 	}
-	if err := open.Start(url.String()); err != nil {
+	if showUrl {
+		c.Ui.Output(fmt.Sprintf("URL for web UI: %s", url.String()))
```

Comment:
This message will cause the URL to be displayed twice:

```console
$ nomad ui -show-url
Opening URL "http://127.0.0.1:4646"
URL for web UI: http://127.0.0.1:4646
```

Perhaps we can delay outputting the message until all conditions are checked? So something like:

```go
var outptut string
if authenticate && ottSecret != "" {
	output = fmt.Sprintf("Opening URL %q with one-time token", url.String()))
	url.RawQuery = fmt.Sprintf("ott=%s", ottSecret)
} else {
	output = fmt.Sprintf("Opening URL %q", url.String()))
}

if showUrl {
	c.Ui.Output(fmt.Sprintf("URL for the Nomad Web UI: %s", url.String()))
	return 0
}

if err := open.Start(url.String()); err != nil {
	c.Ui.Error(fmt.Sprintf("Error opening URL: %s", err))
	return 1
}

c.Ui.Output(output)
return 0
```

What do you think?

Reply:
Ah.. fixed